train.py

```diff
@@ -338,19 +344,118 @@ def evaluate_model_vs_random(model: torch.nn.Module, epsilon: float, num_games:
     win_rate = player_wins / num_games
     draw_rate = draws / num_games
 
-    return win_rate, draw_rate
+    return -win_rate, draw_rate
+
+
+class CheckersQModelWrapper(BaseEstimator, RegressorMixin):
+    def __init__(self, num_hidden_layers=0, hidden_size=0, learning_rate=0.01,
+                 self_play_games_per_iter=25, weight_decay=1e-4):
+        self.num_hidden_layers = num_hidden_layers
+        self.hidden_size = hidden_size
+        self.learning_rate = learning_rate
+        self.self_play_games_per_iter = self_play_games_per_iter
+        self.weight_decay = weight_decay
+
+        # Initialize model here
+        self.model = None
+
+        # Initialize train_hist to an empty list
+        self.train_hist = []
+
+    def fit(self, X, y=None, **kwargs):
+        # Update the model's parameters
+        self.model = CheckersQModel(
+            num_hidden_layers=self.num_hidden_layers,
+            hidden_size=self.hidden_size,
+            output_size=Game.NUM_ACTIONS
+        )
+
+        # Set other parameters as needed
+
+        # Train the model and get the metric to optimize (e.g., average win rate)
+        self.train_hist = train_loop(self.model, learning_rate=self.learning_rate,
+                                     self_play_games_per_iter=self.self_play_games_per_iter,
+                                     weight_decay=self.weight_decay)
```

Comment:
I've made some significant changes to the train.py class, which includes removing the train_loop function. I will create a new branch and update the hyperparameter search code to work with those changes